# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/Shrinidhi-02/flyrank-machine-learning/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

In [10]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.

My lane is Refresh / Content Opportunity Scoring. I want the ML system to help identify which pages should be reviewed first for a possible content refresh. Instead of manually checking every page, I want to use available search and page-performance signals to prioritize pages that may need attention. The goal is to support human review and decision-making, not to automatically decide what should be changed.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

In [11]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


The decision is which pages should be prioritized for human review for a possible content refresh. A content or SEO team member would use the prioritized list to decide which pages to inspect and whether a refresh may be appropriate. A wrong recommendation could waste time by reviewing a page that does not need attention, or cause a page that needs attention to be overlooked. Therefore, the output should support prioritization and human review rather than automatically deciding that a page must be refreshed.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [12]:
import pandas as pd

# Load the FlyRank starter dataset from our GitHub repository
url = "https://raw.githubusercontent.com/Shrinidhi-02/flyrank-machine-learning/main/data/raw/content_refresh_anonymized.csv"

df = pd.read_csv(url)

# Basic dataset size
total_pages = len(df)

# Pages with enough age and search visibility to be considered for refresh
refresh_candidates = df[
    (df["content_age_days"] >= 90) &
    (df["impressions_90d"] > 0)
].copy()

# Count pages showing a downward trend
declining_pages = (refresh_candidates["trend_direction"] == "down").sum()

# Percentage of refresh candidates that are declining
declining_rate = (
    declining_pages / len(refresh_candidates) * 100
    if len(refresh_candidates) > 0 else 0
)

print(f"Total pages: {total_pages:,}")
print(f"Pages eligible for refresh analysis: {len(refresh_candidates):,}")
print(f"Declining pages among these candidates: {declining_pages:,}")
print(f"Declining rate: {declining_rate:.2f}%")


Total pages: 30,000
Pages eligible for refresh analysis: 30,000
Declining pages among these candidates: 16,262
Declining rate: 54.21%


The starter dataset contains 30,000 pages. After applying the refresh-analysis filters of at least 90 days of content age and more than zero impressions in the last 90 days, all 30,000 pages remain eligible. Among these pages, 16,262 are marked as declining, which is 54.21% of the dataset. These numbers show that there is a substantial group of pages whose performance may deserve further investigation, making refresh opportunity prioritization a useful direction for the next stages of the project.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

In [13]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


My analysis can identify observed patterns in the available page and search-performance data and can help prioritize pages for further human review. I can report measured relationships and use the results as decision-support for content refresh prioritization. However, this analysis cannot prove that a particular factor causes a page to decline, guarantee that refreshing a page will improve its performance, or predict Google's future ranking decisions. The results should therefore be treated as directional evidence for human review, not as causal proof or automatic decisions.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.